db/versions/0091-migration.sql

```diff
@@ -0,0 +1,166 @@
+begin
+  -- migration is only possible when all tasks were migrated to new column structure
+  -- following query is commented out for documentation purposes, to prevent failed migration on prod deployment
+  -- IF EXISTS (
+  --   SELECT message_id FROM azure_queue_messages
+  --   WHERE queue_name NOT IN ('claim-queue', 'deadline-queue', 'resolved-queue')
+  --   AND task_queue_id IS NULL
+  --   LIMIT 1
+  -- ) THEN
+  --   RAISE EXCEPTION 'Not possible to migrate, some records are still in old format';
+  -- END IF;
+
+  -- migration of data
+  -- prevent reads and writes from table, get exclusive access
+  LOCK TABLE azure_queue_messages;
+
+  -- Task Deadlines
+  -- purpose: know when particular task expires, so dependencies/task groups can be resolved
+  CREATE TABLE queue_task_deadlines (
+    task_id text not null,
+    task_group_id text not null,
+    scheduler_id text not null,
+    created timestamptz not null,
+    deadline timestamptz not null,
+    visible timestamptz not null,
```

Comment:
I take back my comment, sorry :) 
I realized all methods are using `pop_receipt + visible` fields to "check-out" messages and make sure no one is working on them concurrently. So if I'd do this with `deadline`, `DeadlineResolver` would fail to acknowledge and process this entry, because original deadline defined in task would not match.

Reply:
Good spot - that makes total sense!